### DISCUSSION TITLE
Encryption and Decryption of PDFs examples in documentation remove attachments, links, ...

### DISCUSSION THREAD

**Discussion by redfast00** [py-pdf/pypdf#2544] [Q&A] [answered]
I want to decrypt a PDF document, while keeping the document intact. The documentation gives an example for this (https://pypdf.readthedocs.io/en/stable/user/encryption-decryption.html); but this decrypts the document, then reconstructs it by copying page by page to the new document.

This removes attachments, but also the table of contents, the links in the document pointing to other pages, ...

I would like an approach where the encrypted data is decrypted 'in-place', keeping the document structure intact.

## Environment

Which environment were you using when you encountered the problem?

```bash
$ python -m platform
Linux-6.5.0-26-generic-x86_64-with-glibc2.35

$ python -c "import pypdf;print(pypdf._debug_versions)"
pypdf==4.1.0, crypt_provider=('cryptography', '42.0.5'), PIL=none
(modified by me to add support for the pubsec decryptor, but this doesn't affect anything)
```

## Code + PDF

This is a minimal, complete example that shows the issue; it does not use encryption, but it has the same problem as the sample that does use encryption.

```python
from pypdf import PdfReader, PdfWriter

reader = PdfReader("PN7160_PN7161.pdf")
writer = PdfWriter()

for idx, page in enumerate(reader.pages):
    writer.add_page(page)

print("Saving to file...")
# Save the new PDF to a file
with open("out.pdf", "wb") as f:
    writer.write(f)
```
[PN7160_PN7161.pdf](https://github.com/py-pdf/pypdf/files/14756976/PN7160_PN7161.pdf)

**Comment by j-t-1** [comment, marked as answer]
Create a backup of the PDF, then switch these:

```
writer = PdfWriter()

for idx, page in enumerate(reader.pages):
    writer.add_page(page)
```
`writer = PdfWriter(clone_from=reader)`

**Comment by redfast00** [comment]
@j-t-1 that works! Should I make a PR to replace that in the documentation?

**Comment by j-t-1** [comment]
Could do, currently it is not obvious that adding each page is insufficient to have equivalency; I had a similar problem #2485.
